platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheParallelStoreIntegrationTest.groovy

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.cc.impl
+
+import org.gradle.test.fixtures.file.TestFile
+
+class ConfigurationCacheParallelStoreIntegrationTest extends AbstractConfigurationCacheIntegrationTest {
+
+    def "failures while storing different projects are reported"() {
+        given:
+        def projects = (1..5).collect  {"proj$it" }
+        def configurationCache = newConfigurationCacheFixture()
+        settingsFile << """
+            include ${projects.collect {"'${it}'" }.join(',')}
+        """
+
+        def createBuildFile = {
+            file(getDefaultBuildFileName()) << """
+                task t {
+                    outputs.file({ -> throw new RuntimeException("\${ project.identityPath } went BOOM!") })
+                }
+            """
+        }
+
+        projects.each {
+            createDir(it, createBuildFile)
+        }
+
+        when:
+        configurationCacheFails("t")
+
+        then:
+        configurationCache.assertStateStoreFailed()
+        outputContains("Configuration cache entry discarded due to serialization error.")
+        failure.assertHasFailures(1)
+        // we cannot assert on the build script absolute path as it is non-deterministic
+        failure.assertHasLineNumber(3)
+        failure.assertHasDescription("Error while saving task graph")
```

Comment:
Calling it "task graph" even though it is not exact, as our docs don't mention the concept of "work graph"